### PR TITLE
ci: support sle-micro for hal cluster

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9691

#### What this PR does / why we need it:

support sle-micro for hal cluster

## Summary by CodeRabbit

- **New Features**
	- Introduced a new Terraform configuration for managing a Rancher2 environment with Harvester integration.
	- Added several new variables for enhanced configuration and sensitive information handling in Terraform.

- **Bug Fixes**
	- Simplified error messages related to unsupported operating systems in the setup scripts.
	- Streamlined conditional checks in the setup scripts to improve control flow.

- **Documentation**
	- Enhanced variable descriptions in the Terraform configuration for better clarity.

### CODE REVIEW

File: test_framework/terraform/harvester/sle-micro/main.tf [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      snapshot_schedule_cron = "0 */5 * * *"
+      snapshot_retention = 5
+    }
+    chart_values = ""
+  }
+}
+
+output "kube_config" {
+  value = rancher2_cluster_v2.e2e-cluster.kube_config
+  sensitive = "true"
```
[the remark this answers is not in the document]
Comment (khushboo-rancher) [reply test_framework/terraform/harvester/sle-micro/main.tf:208]: @yangchiu Looks like a reasonable review comment.

**Review by khushboo-rancher — COMMENTED** [review]
LGTM except one comment from coderabbit looks legit.

File: test_framework/terraform/harvester/sle-micro/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (yangchiu) [reply test_framework/terraform/harvester/sle-micro/main.tf]: Refined.

**Review by roger-ryao — APPROVED** [review]
LGTM

**Pull request merged**